components/cdc/src/endpoint.rs

```diff
@@ -1334,16 +1338,26 @@ impl<T: 'static + CdcHandle<E>, E: KvEngine, S: StoreRegionMeta + Send> Runnable
 
 pub struct CdcTxnExtraScheduler {
     scheduler: Scheduler<Task>,
+    memory_quota: Arc<MemoryQuota>,
 }
 
 impl CdcTxnExtraScheduler {
-    pub fn new(scheduler: Scheduler<Task>) -> CdcTxnExtraScheduler {
-        CdcTxnExtraScheduler { scheduler }
+    pub fn new(scheduler: Scheduler<Task>, memory_quota: Arc<MemoryQuota>) -> CdcTxnExtraScheduler {
+        CdcTxnExtraScheduler {
+            scheduler,
+            memory_quota,
+        }
     }
 }
 
 impl TxnExtraScheduler for CdcTxnExtraScheduler {
     fn schedule(&self, txn_extra: TxnExtra) {
+        let size = txn_extra.size();
+        if let Err(e) = self.memory_quota.alloc(size) {
+            warn!("cdc schedule txn extra failed on alloc memory quota";
```

Comment:
It may overwhelm tikv logs. How about changing to debug level and record such error in metric?